src/Widgets/NotificationEntry.vala

```diff
@@ -158,17 +168,27 @@ public class Notifications.NotificationEntry : Gtk.ListBoxRow {
 
         deck.notify["visible-child"].connect (() => {
             if (deck.transition_running == false && deck.visible_child != grid) {
+                dismiss ();
                 remove_notification_entry ();
             }
         });
 
         deck.notify["transition-running"].connect (() => {
             if (deck.transition_running == false && deck.visible_child != grid) {
+                dismiss ();
                 remove_notification_entry ();
             }
         });
     }
 
+    public void dismiss () {
+        revealer.reveal_child = false;
+        GLib.Timeout.add (revealer.transition_duration, () => {
```

Comment:
`notify["child-revealed"]`

Reply:
Well I mean yeah if you wanna do things the easy way duh. I knew that :p